R/data-combined.R

```diff
@@ -653,42 +650,30 @@ DataCombined <- R6::R6Class(
       # raw data that needs to be transformed.
       dataArg <- dplyr::tibble(
         name          = forNames %||% unique(data$name),
-        xOffsets      = xOffsets,
-        yOffsets      = yOffsets,
-        xScaleFactors = xScaleFactors,
-        yScaleFactors = yScaleFactors
+        # For offsets: `0` (default for no change)
+        xOffsets      = tidyr::replace_na(xOffsets, 0),
+        yOffsets      = tidyr::replace_na(yOffsets, 0),
+        # For scale factors: `1` (default for no change)
+        xScaleFactors = tidyr::replace_na(xScaleFactors, 1),
+        yScaleFactors = tidyr::replace_na(yScaleFactors, 1)
       )
 
-      # Update data frame using given transformation parameters
-      private$.dataTransformations <- private$.updateDataFrame(private$.dataTransformations, dataArg)
+      # Update private$.dataTransformations with new transformation settings
+      private$.dataTransformations <-
+        private$.updateDataFrame(private$.dataTransformations,
+                                 dataArg) %>%
+        # Preserve the order from data
+        dplyr::arrange(match(name, data$name))
 
       # Every call to method to set transformations refreshes these parameters.
       #
       # Thus, if there are any existing parameters from object's lifecycle,
-      # they should be removed.
-      data <- dplyr::select(data, -dplyr::ends_with(c("Offsets", "ScaleFactors")))
-
-      # Datasets for which no data transformations were specified, there will be
-      # missing values, which need to be replaced by defaults for no change.
-      data <- dplyr::left_join(data, private$.dataTransformations, by = "name")
-
-      # For offsets: `0` (default for no change)
-      data <- dplyr::mutate(
-        data,
-        dplyr::across(
-          .cols = matches("offsets$"), # relevant only for columns matching this pattern
-          .fns = function(x) tidyr::replace_na(x, 0)
-        )
-      )
-
-      # For scale factors: `1` (default for no change)
-      data <- dplyr::mutate(
-        data,
-        dplyr::across(
-          .cols = matches("scalefactors$"), # relevant only for columns matching this pattern
-          .fns = function(x) tidyr::replace_na(x, 1)
-        )
-      )
+      # they should be removed and set with the new/default values.
+      data <-
+        data %>%
+        dplyr::select(-tidyr::any_of(colnames(private$.dataTransformations)[-1])) %>%
+        dplyr::left_join(private$.dataTransformations,
+                         by = "name")
```

Comment:
same with this. line 705, the arrange method is called and it is organized differently. I think we need to write the code one way or the other. In the same PR, we HAVE to be consistent . Also specifying named parameter onkly for one parameter..is this usual in R? Why not for the FIRST PARAMETER?

Reply:
I used {styler} that autoformatted this. I will make it consistend in next commits.

And yes, usually, first and sometimes second arguments are not named.
> In general, use positional matching only for the first one or two arguments; they will be the most commonly used, and most readers will know what they are. Avoid using positional matching for less commonly used arguments/
> source: https://adv-r.hadley.nz/functions.html?q=functions#prefix-form